dist/src/main/java/io/camunda/application/commons/console/ping/PingConsoleRunner.java

```diff
@@ -169,16 +184,38 @@ private Either<Exception, String> getLicensePayload() {
     final LicensePayload payload =
         new LicensePayload(
             license,
-            clusterId,
+            brokerTopologyManager
+                .getClusterConfiguration()
+                .clusterId()
+                .orElseThrow(() -> new IllegalStateException("Cluster ID is not set.")),
             pingConfiguration.clusterName(),
             VersionUtil.getVersion(),
             getActiveProfiles(),
             pingConfiguration.properties());
     try {
-      return Either.right(objectMapper.writeValueAsString(payload));
+      licensePayload = Either.right(objectMapper.writeValueAsString(payload));
     } catch (final JsonProcessingException exception) {
-      return Either.left(exception);
+      licensePayload = Either.left(exception);
+    }
+  }
+
+  private CompletableFuture<Void> waitForClusterId() {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+    if (brokerTopologyManager.getClusterConfiguration().clusterId().isPresent()) {
```

Comment:
🔧 We could transfer the ClusterId during the Future completion, and then pass it in the license building payload, then we dont need to check/validate the topology manager anymore. Just a thought.